src/sqlancer/cnosdb/CnosDBLoggableFactory.java

```diff
@@ -0,0 +1,59 @@
+package sqlancer.cnosdb;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import sqlancer.cnosdb.query.CnosDBOtherQuery;
+import sqlancer.cnosdb.query.CnosDBQueryAdapter;
+import sqlancer.common.log.Loggable;
+import sqlancer.common.log.LoggableFactory;
+import sqlancer.common.log.LoggedString;
+import sqlancer.common.query.ExpectedErrors;
+import sqlancer.common.query.Query;
+
+public class CnosDBLoggableFactory extends LoggableFactory {
+
+    @Override
+    protected Loggable createLoggable(String input, String suffix) {
+        String completeString = input;
+        if (!input.endsWith(";")) {
+            completeString += ";";
+        }
+        if (suffix != null && suffix.length() != 0) {
+            completeString += suffix;
+        }
+        return new LoggedString(completeString);
+    }
+
+    @Override
+    public CnosDBQueryAdapter getQueryForStateToReproduce(String queryString) {
+        ExpectedErrors errors = new ExpectedErrors();
+        errors.addAll(CnosDBExpectedError.expectedErrors());
+        return new CnosDBOtherQuery(queryString, errors);
+    }
+
+    @Override
+    public CnosDBQueryAdapter commentOutQuery(Query<?> query) {
+        String queryString = query.getLogString();
+        String newQueryString = "-- " + queryString;
+        ExpectedErrors errors = new ExpectedErrors();
+        errors.addAll(CnosDBExpectedError.expectedErrors());
```

Comment:
I believe that commenting out a SQL statement should not result in any errors, so the expected errors could be empty here?